lib/Model/Binary.cpp

```diff
@@ -95,6 +100,8 @@ bool Binary::verifyTypes(VerifyHelper &VH) const {
 }
 
 void Binary::dump() const {
+  revng::Tracking::push(*this);
```

Comment:
I think we should use `TrackGuard` here. Just pushing is a bug.

Reply:
You have a well self-contained method (`dump`) that pushes stuff but does not pop, how is this correct? o_O